### PR TITLE
feature: Added ability to use Escoria variables with the custom command

### PR DESCRIPTION
**Pull request by balloonpopper**
It annoyed me that after the change to allow parameters to be passed to "custom" that I couldn't pass Escoria variables.
Fixes : https://github.com/godot-escoria/escoria-issues/issues/261

### CODE REVIEW

**Review by StraToN — APPROVED** [review]

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd [hunk 1]
```diff
@@ -81,6 +81,11 @@ func run(command_params: Array) -> int:
 	var object = escoria.object_manager.get_object(
 		command_params[0]
 	)
+	# Global variables can be substituted into the command arguments by wrapping the global
+	# name in braces.
+	for loop in command_params[3].size():
+		command_params[3][loop] = escoria.logger.replace_globals(command_params[3][loop])
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc/commands/custom.gd:87]: If we're going to be replacing globals for things other than log-related commands, then I'd like to see `replace_globals` extracted from `logger` and put into a more appropriate place, e.g. `ESCCompiler`. @StraToN would you agree with that location? Or would you prefer it put somewhere else?

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/core-scripts/esc/commands/custom.gd (hunk 1 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/commands/custom.gd:87]: I totally agree, `replace_globals` will move out of ESCLogger.

**Comment by balloonpopper** [comment]
@StraToN @BHSDuncan I've moved the globals_replace function to esc_compiler.
Tested the print, say and custom functions

**Review by BHSDuncan — APPROVED** [review]

**Review by StraToN — APPROVED** [review]

**Comment by StraToN** [comment]
Looks good to me. Thanks a lot!

**Pull request merged**